File: cs/Iris/ParserUtils/LexParseException/BaseException.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Iris.ParserUtils.LexParseException
{
    public abstract class BaseException : Exception
    {
        protected int lineNum;
        protected int columnNum;

        public BaseException(int lineNumber, int columnNumber)
        {
            this.lineNum = lineNumber;
            this.columnNum = columnNumber;
        }

        public string stringTrace()
        {
            StringBuilder str = new StringBuilder();

            str.Append("line ");
            str.Append(lineNum);
            str.Append(", char ");
            str.Append(columnNum);
            str.Append(' ');

            return str.ToString();
        }

        public abstract string ExceptionToString();
    }
}

File: java/Iris/src/com/rokuan/iris/exception/BaseException.java
package com.rokuan.iris.exception;

public abstract class BaseException extends Exception
{
	protected int lineNum;
	protected int columnNum;

	public BaseException(int lineNumber, int columnNumber)
	{
		this.lineNum = lineNumber;
		this.columnNum = columnNumber;
	}

	public String stringTrace()
	{
		StringBuilder str = new StringBuilder();

		str.append("line ");
		str.append(lineNum);
		str.append(", char ");
		str.append(columnNum);
		str.append(' ');

		return str.toString();
	}

	public abstract String ExceptionToString();

	public String toString(){
		return this.ExceptionToString();
	}
}
